andela/ah-magnificent6
# #[159965302] Email verification upon registration

**What does this PR do?**
Have an email sent to the user once they register. The email contains an activation link which when clicked activates the user.

**Description of Task to be completed?**
An Email should be sent to a user upon successful registration. User should be able to receive a verification link via email upon successful signup

**How should this be manually tested?**
Run `git clone https://github.com/andela/ah-magnificent6.git` from your local machine to clone the repo. 
After cloning the repo, make the repo folder your current working directory. Run `git checkout ft-email-verification-159965302` to go to the relevant branch. 
Create a virtual environment by running `python3 -m venv venv` and `source venv/bin/activate` to activate it. After activating it, run `pip install -r requirements.txt` to install the required packages.
Create a `.env` file from the `.env-sample` and run `python manage.py runserver` to start the app.
Add email configuration settings in your `.env` file : `EMAIL_HOST='host.host.com'
EMAIL_PORT='111'
EMAIL_HOST_USER=from@from.com    
EMAIL_PASSWORD='password'`

Go to the link [http://127.0.0.1:8000/api/user/signup](http://127.0.0.1:8000/api/user/signup) on postman.
The expected payloads are in the format:
User registration: `POST /api/user/signup`
```
{
    "email": "youremailadress@user.user",
    "username": "author",
    "password": "goodpassword"
}
```
Register and check your email address, then click the link and you shall recieve a message that you have been activated. 
You can now log in.

**What are the relevant pivotal tracker stories?**
[#159965302](https://www.pivotaltracker.com/story/show/159965302
<img width="1605" alt="screen shot 2018-09-04 at 16 49 50" src="https://user-images.githubusercontent.com/39065274/45091009-cc055380-b119-11e8-8dc2-7329b12246ac.png">
)
